tuna/fin_class.py

```diff
@@ -291,62 +317,68 @@ def dump_json(self, outfile, to_file=True):
 
     return True
 
-  def run(self, outfile=None, to_file=True):
+  def prep_fin_input(self, outfile=None, to_file=True):
     """Main function in Fin that produces Fin input file"""
 
     self.cnx = self.machine.connect(self.chk_abort_file())
-    ret = ''
+    ret = False
     if outfile is None:
       outfile = "fin_input.json"
     if self.create_dumplist():
       ret = self.dump_json(outfile, to_file)
     else:
-      self.logger.error("Could not create dumplist for Fin input file")
+      self.logger.warning("Could not create dumplist for Fin input file")
 
     return ret
 
+  def insert_applicability(self, session, json_in):
+    """write applicability to sql"""
+    _, solver_id_map_h = get_solver_ids()
+    for elem in json_in:
+      if "applicable_solvers" in elem.keys():
+        #remove old applicability
+        query = 'delete from {} where config={} and session={}'.format(
```

Comment:
Can we turn this delete in SQLAlchemy format?

Reply:
sure